Assets/Plugins/Clean/Editor/Util.cs

```diff
@@ -0,0 +1,190 @@
+                case InputField.ContentType.EmailAddress: return TMP_InputField.ContentType.EmailAddress;
+                case InputField.ContentType.IntegerNumber: return TMP_InputField.ContentType.IntegerNumber;
+                case InputField.ContentType.Name: return TMP_InputField.ContentType.Name;
+                case InputField.ContentType.Password: return TMP_InputField.ContentType.Password;
+                case InputField.ContentType.Pin: return TMP_InputField.ContentType.Pin;
+                case InputField.ContentType.Standard: return TMP_InputField.ContentType.Standard;
+                default: return TMP_InputField.ContentType.Standard;
+            }
+        }
+        
+        public static TMP_InputField.InputType GetTMPInputType( InputField.InputType inputType )
+        {
+            switch( inputType )
+            {
+                case InputField.InputType.AutoCorrect: return TMP_InputField.InputType.AutoCorrect;
+                case InputField.InputType.Password: return TMP_InputField.InputType.Password;
+                case InputField.InputType.Standard: return TMP_InputField.InputType.Standard;
+                default: return TMP_InputField.InputType.Standard;
+            }
+        }
+
+        public static TMP_InputField.LineType GetTMPLineType( InputField.LineType lineType )
+        {
+            switch( lineType )
+            {
+                case InputField.LineType.MultiLineNewline: return TMP_InputField.LineType.MultiLineNewline;
+                case InputField.LineType.MultiLineSubmit: return TMP_InputField.LineType.MultiLineSubmit;
+                case InputField.LineType.SingleLine: return TMP_InputField.LineType.SingleLine;
+                default: return TMP_InputField.LineType.SingleLine;
+            }
+        }
+
+        // TODO: Investigate if there is/should be a public API for this
+        private static FieldInfo GetUnityPersistentCalls()
+        {
+            var calls = typeof(UnityEventBase).GetField("m_PersistentCalls",
+                BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+            if (calls != null) return calls;
+            
+            var listeners = typeof(UnityEventBase).GetField("m_PersistentListeners",
+                BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+            if (listeners!= null) return listeners;
+
+            throw new Exception("Could not retrieve m_PersistentCalls to copy Unity Events");
+        }
+
+        public static object CopyUnityEvent(UnityEventBase target)
+        {
+            var field = GetUnityPersistentCalls();
+            return field.GetValue(target);
+        }
+
+        public static void PasteUnityEvent(UnityEventBase target, object unityEvent)
+        {
+            var field = GetUnityPersistentCalls();
+            field.SetValue(target, unityEvent);
+        }
```

Comment:
Currently this does *not* work, and there is a failing test for it.  I am unsure if the owner of the asset is aware of this 😅

Reply:
Any idea what's wrong there? I see it's using Reflection to get the field and set it again, but what's failing? Do you get an exception?

Reply:
Not a clue what's failing.  No exceptions, just silent.
This is the test - https://github.com/iambalaam/unity-text-to-tmp/blob/main/Assets/Plugins/Clean/Tests/UnityEventTests.cs

Maybe my understanding of how it should work is incorrect?